src/gtk/window.cpp

```diff
@@ -2702,20 +2702,63 @@ bool wxWindowGTK::Create( wxWindow *parent,
         m_widget = m_wxwindow;
     else
         GTKCreateScrolledWindowWith(m_wxwindow);
-    g_object_ref(m_widget);
-
-    if (m_parent)
-        m_parent->DoAddChild( this );
 
     m_focusWidget = m_wxwindow;
 
     SetCanFocus(AcceptsFocus());
 
     PostCreation();
 
+    GTKEnableOverlay();
+
+    if (m_parent)
+        m_parent->DoAddChild( this );
+
+    g_object_ref(m_widget);
+
     return true;
 }
 
+void wxWindowGTK::GTKEnableOverlay()
+{
+#ifdef __WXGTK3__
+    if ( !wxGetEnv(wxS("WXUSE_OVERLAY"), NULL) )
+        return;
+
+    if ( IsTopLevel() || IsOfStandardClass() )
+        return;
+
+    if ( GTK_IS_OVERLAY(m_widget) ||
+         GTK_IS_OVERLAY(gtk_widget_get_parent(m_wxwindow)) )
+        return;
+
+    GtkWidget* overlay = gtk_overlay_new();
+
+    if ( m_widget != m_wxwindow )
+    {
+        wxLogDebug("GTKEnableOverlay( m_widget != m_wxwindow )");
+
+        g_object_ref(m_wxwindow);
+        gtk_container_remove( GTK_CONTAINER(m_widget), m_wxwindow );
+
+        gtk_container_add( GTK_CONTAINER(overlay), m_wxwindow );
+        gtk_container_add( GTK_CONTAINER(m_widget), overlay );
+    }
+    else // m_widget == m_wxwindow
+    {
+        wxLogDebug("GTKEnableOverlay( m_widget == m_wxwindow )");
+
+        gtk_container_add( GTK_CONTAINER(overlay), m_widget );
+
+        m_widget = overlay;
+    }
+
+    gtk_widget_show_all(m_widget);
```

Comment:
We didn't do this before, her,e but only in `PostCreation()`. Does it really need to be done here now? If so, it should at least check `m_isShown`, as the existing call does.